authority/tls.go

```diff
@@ -645,6 +645,85 @@ func (a *Authority) GetTLSCertificate() (*tls.Certificate, error) {
 	return &tlsCrt, nil
 }
 
+// GetTLSClientCertificate creates a new leaf certificate to be used for webhook
```

Comment:
Mostly copied from https://github.com/smallstep/certificates/pull/1001/commits/a30444b0d4c3f0af514b428f59d2989cf9b5b358#diff-664ff05b55cabc1ecdc81445fa819245e11fc531aa6b25a0fa6b2d737176d558R569

Reply:
I think we can reuse the server tls.Config for this.